pkg/ingester/ingester_v2.go

```diff
@@ -29,9 +30,18 @@ const (
 	errTSDBCreateIncompatibleState = "cannot create a new TSDB while the ingester is not in active state (current state: %s)"
 )
 
+// UserTSDB is a users TSDB
+type UserTSDB struct {
+	sync.Mutex
```

Comment:
If this is just used to lock around `lastWritten` I would be more explicit and name it `lastWrittenLock`. If in the future we'll use it for more purposes, we'll rename it.

Reply:
Agreed, changed.